### ISSUE TITLE
CA1018: False positive on attribute that inherits another attribute

### ISSUE THREAD

**Issue by reduckted** [dotnet/roslyn-analyzers#1732]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.6.1

#### Diagnostic ID

[CA1018](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1018-mark-attributes-with-attributeusageattribute)

#### Repro steps

1. Create two attributes, `FooAttribute` and `BarAttribute`.
2. Make `BarAttribute` inherit from `FooAttribute`.
3. Only specify `AttributeUsage` on `FooAttribute`.

Example:

```c#
[AttributeUsage(AttributeTargets.Class, Inherited = true)]
class FooAttribute : Attribute
{
}

class BarAttribute : FooAttribute
{
}
```

#### Expected behavior

The analyzer does not produce any warnings.

#### Actual behavior

The analyzer produces "CA1018: Specify AttributeUsage on BarAttribute". 

Specifying `AttributeUsage` on `BarAttribute` is unnecessary because it inherits the usage from `FooAttribute`. This can be proven with the following code snippet:

```c#
[Foo()] // No error here.
[Bar()] // No error here.
class Test
{
    [Foo()] // CS0592 Attribute 'Foo' is not valid on this declaration type. It is only valid on 'class' declarations.
    [Bar()] // CS0592 Attribute 'Bar' is not valid on this declaration type. It is only valid on 'class' declarations.
    void Method()
    {
    }
}
```

The same thing happens for VB.NET code.

**Comment by sharwell** [comment]
👍 I've observed this pattern in some simplification cases as well, and the special case seems to work well with [CA1813 (Avoid unsealed attributes)](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1813-avoid-unsealed-attributes).

**Comment by mavasani** [comment]
This diverges from the FxCop behavior, but I agree that the proposed change is reasonable.

**Comment by sharwell** [comment]
@mavasani I encountered the issue above today in the 5.0.3 release. I see pull request #1764 linked to this issue, but that pull request does not address the issue reported here.
